src/main/java/dev/codesoapbox/dummy4j/ExpressionResolver.java

```diff
@@ -9,13 +9,24 @@
      * Resolves an expression like:
      * {@code #{name.male_first_name} #{name.last_name} }
      * <p>
-     * Definition keys are defined as {@code #{definition.path}}.
+     * Single-locale placeholders are defined as {@code #{definition.path}}.
+     * They resolve to a random value from a single locale.
+     * <p>
+     * Multi-locale placeholders (since SNAPSHOT) are defined as {@code #{{definition.path}}}.
+     * They resolve to a random value from a superset of all locales' values.
      * <p>
      * Digits are defined simply as {@code #}.
      * <p>
-     * Placeholders which have no definitions will be removed.
-     * The method first looks into the locale which was passed as first in the list and then goes to the next
-     * ones in order if a key could not be resolved.
+     * Placeholders which don't resolve to anything will be removed.
```

Comment:
Removed from where? 
What is the difference between a `placeholder which don't resolve` and `a path that exists but does not resolve to a value or a list`?

Is it something like the `shouldReturnEmptyStringWhenUnableToResolveSingleLocalePlaceholder()` test or the `shouldReturnEmptyStringWhenResolvedPathIsEmpty()` test?

Reply:
1. Removed from the expression that is being resolved. This means they will essentially be replaced with an empty string (rather than, e.g. "null").
2. `a path that exists but does not resolve to a value or a list` is basically a path to keys, rather than a path to values. Perhaps it would be better to write `if a path resolves to a list of keys, returns a random key contained directly within it.`?
3. I think both tests test this.

Unrelated - might want to check how resolving `#{key1}.` (with a dot at the end) will work.

Reply:
I tried to add the following line to my project:
```
System.out.println(dummy.expressionResolver().resolve("#{book.}"));
```

And got:
```
2021-04-12 15:05:00.158 ERROR 15448 --- [nio-8080-exec-1] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0] with root cause

java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at dev.codesoapbox.dummy4j.definitions.LocalizedDummyDefinitionsMap.resolve(LocalizedDummyDefinitionsMap.java:38) ~[dummy4j-0.7.3-SNAPSHOT.jar:0.7.3-SNAPSHOT]
	at dev.codesoapbox.dummy4j.definitions.LocalizedDummyDefinitionsMap.resolveResult(LocalizedDummyDefinitionsMap.java:49) ~[dummy4j-0.7.3-SNAPSHOT.jar:0.7.3-SNAPSHOT]
	at dev.codesoapbox.dummy4j.definitions.LocalizedDummyDefinitionsMap.resolve(LocalizedDummyDefinitionsMap.java:42) ~[dummy4j-0.7.3-SNAPSHOT.jar:0.7.3-SNAPSHOT]
	at dev.codesoapbox.dummy4j.definitions.LocalizedDummyDefinitionsMap.resolve(LocalizedDummyDefinitionsMap.java:34) ~[dummy4j-0.7.3-SNAPSHOT.jar:0.7.3-SNAPSHOT]
	at dev.codesoapbox.dummy4j.DefaultExpressionResolver.resolveKey(DefaultExpressionResolver.java:100) ~[dummy4j-0.7.3-SNAPSHOT.jar:0.7.3-SNAPSHOT]
	at dev.codesoapbox.dummy4j.DefaultExpressionResolver.lambda$replaceKeyPlaceholders$1(DefaultExpressionResolver.java:66) ~[dummy4j-0.7.3-SNAPSHOT.jar:0.7.3-SNAPSHOT]
	at dev.codesoapbox.dummy4j.DefaultExpressionResolver.replace(DefaultExpressionResolver.java:82) ~[dummy4j-0.7.3-SNAPSHOT.jar:0.7.3-SNAPSHOT]
	at dev.codesoapbox.dummy4j.DefaultExpressionResolver.replaceKeyPlaceholders(DefaultExpressionResolver.java:65) ~[dummy4j-0.7.3-SNAPSHOT.jar:0.7.3-SNAPSHOT]
	at dev.codesoapbox.dummy4j.DefaultExpressionResolver.resolveAllKeysAndDigits(DefaultExpressionResolver.java:58) ~[dummy4j-0.7.3-SNAPSHOT.jar:0.7.3-SNAPSHOT]
	at dev.codesoapbox.dummy4j.DefaultExpressionResolver.resolve(DefaultExpressionResolver.java:49) ~[dummy4j-0.7.3-SNAPSHOT.jar:0.7.3-SNAPSHOT]
```

Reply:
Thanks, added the tests to the branch.